doc/develop/test/ztest.rst

```diff
@@ -458,6 +458,30 @@ Example output for a failed macro from
 
 .. doxygengroup:: ztest_assert
 
+
+Expectations
+============
+
+These macros will continue test execution if the related expectation fails.  When an expectation
+fails, it will print the current file, line, and function, alongside a reason for the failure and an
+optional message but continue executing the test. If the config option:`CONFIG_ZTEST_ASSERT_VERBOSE`
+is 0, the expectations will only print the file and line numbers, reducing the binary size of the
+test.
+
+Example output for a failed macro from::
+
+  zexpect_equal(buf->ref, 2, "Invalid refcount");
+  zexpect_equal(buf->ref, 1337, "Invalid refcount");
+
```

Comment:
how to use zepxect_fail, seeing from the code, it needs a special call.

Reply:
`ZTEST_EXPECT_FAIL/SKIP` are macros that validate a specific test has failed. It was merged back in https://github.com/zephyrproject-rtos/zephyr/commit/84dfb8edf86bc26c80cb2b8987c8a3dc164c803b and is not part of this PR. However, if we find this highly confusing we can change the name of `zexpect` to something else.

Reply:
I think @hakehuang may have been asking for an example of `ZTEST_EXPECT_FAIL` here.